File: cpp_version/5.cpp
#include <iostream>

class Frog
{
private:
    int x, y;

public:
    Frog(int x, int y) : x(x), y(y) {}
    void moveLeft() { x--; }
    void moveRight() { x++; }
    void moveUp() { y++; }
    void moveDown() { y--; }
    int getCurrentPositionX() { return x; }
    int getCurrentPositionY() { return y; }
    void print()
    {
        std::cout << x << " " << y << std::endl;
    }
};

int main()
{
    int x, y, finalX, finalY;
    std::cin >> x >> y;
    Frog frog(x, y);
    std::cin >> finalX >> finalY;
    while ((frog.getCurrentPositionX() != finalX) || (frog.getCurrentPositionY() != finalY))
    {
        if (frog.getCurrentPositionX() < finalX)
            frog.moveRight();
        else if (frog.getCurrentPositionX() > finalX)
            frog.moveLeft();
        if (frog.getCurrentPositionY() < finalY)
            frog.moveUp();
        else if (frog.getCurrentPositionY() > finalY)
            frog.moveDown();
    }
    frog.print();

    return 0;
}

File: py_version/5.py
class Frog:
    def __init__(self, x, y):
        self.x = x
        self.y = y
    def moveLeft(self):
        self.x -= 1
    def moveRight(self):
        self.x += 1
    def moveUp(self):
        self.y += 1
    def moveDown(self):
        self.y -= 1
    def getCurrentPositionX(self):
        return self.x
    def getCurrentPositionY(self):
        return self.y
    def displayCurrentPosition(self):
        print(f"({self.x}, {self.y})")

x, y, dist_x, dist_y = map(int, input().split())
frog = Frog(x, y)
while (frog.getCurrentPositionX() != dist_x) or (frog.getCurrentPositionY() != dist_y):
    if frog.getCurrentPositionX() < dist_x:
        frog.moveRight()
    elif frog.getCurrentPositionX() > dist_x:
        frog.moveLeft()
    if frog.getCurrentPositionY() < dist_y:
        frog.moveUp()
    elif frog.getCurrentPositionY() > dist_y:
        frog.moveDown()
    frog.displayCurrentPosition()
# Input: current position of the frog, destination position of the frog
# Output: the path the frog takes to reach the destination position
    
